docs/configuration.rst

```diff
@@ -78,49 +78,46 @@ The JBoss EAP artifact will be fetched from: ``http://cache.host.com/cache/jboss
 
     In all cases digest will be computed from the downloaded file and compared with the expected value.
 
-.. _configuration_repositories:
+:: _repo_inject:
 
-``[repositories]``
--------------------
-
-The ``repository`` section can be used to define YUM/DNF repository files
-that should be added to the image at build time.
+ Repository injection
+^^^^^^^^^^^^^^^^^^^
+There are multiple possibilities how to inject repository inside the image and its configurable in the ``~/.cekit/config``. The configuration follows this scheme:
 
 .. code::
 
-    [repositories]
-    jboss-ocp=http://host/jboss-rhel-os.repo,http://host/jboss-rhel-ocp.repo,http://host/jboss-rhel-rhscl.repo
-    other-repos=http://otherhost.com/osme.repo
+   [repositories-$builder]
+   $repo_name = $action
 
-In case you have YUM/DNF repo files that you want to put to the ``/etc/yum.repos.d`` directory to enable additional
-repositories Cekit can handle it for you automatically.
+Where:
 
-Repositories are defined using custom keys that could be later referenced in the :ref:`image descriptor's
-packages section <descriptor_packages>`, under ``repositories``.
+* ``$builder`` - one of supported builder (docker, osbs)
 
-.. code:: yaml
+* ``$repo_name`` - name of the repository
+
+* ``$action`` - one of the following actions:
+
+  1) **odcs-pulp** - default for OSBS builder
 
-    packages:
-        repositories:
-            - jboss-ocp
-        install:
-            - mongodb24-mongo-java-driver
-            - postgresql-jdbc
-            - mysql-connector-java
-            - maven
-            - hostname
+     Aks odcs to create a temporary repository from an existing pulp repository. Repository name is taken from
+     ``repository`` atribute of the repository section.
 
-Cekit will copy all repo files to ``/etc/yum.repos.d`` directory and
-enable them to be used while installing packages listed in the packages section.
+  2) **rpm**
 
-At the end of the image build process Cekit removes newly added repo files from the ``/etc/yum.repos.d``
-directory automatically. If you do not want to have these files removed after installation --
-you need to make your repo files part of some module that installs them in the correct place.
+     Install repository from rpm via yum. Mostly used for epel/scl in Centos. RPM name is taken from ``repository``
+     attribute of the repository section.
 
-There are a few rules about repositories added that way:
+  3) **dummy** - default for Docker builder
+
+     Threats the repository as only documentation note, that this repo should be available in the image. It
+     does not inject/enable it in the image.
+
```

Comment:
I think we should add a possibility to specify the repo as a repo file, similarly to what we do currently:

```
[repositories-docker]
rhel-7-server-rpms = http://host.com/composes/rhel-7-server-rpms.repo
```

I think this would be pretty useful for local development.

Reply:
if so.. I would change the object a little. can we have a BJ about it?